pkg/image/importer/importer.go

```diff
@@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
 	}
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
+	bs := repo.Blobs(ctx)
+
+	layerSet := sets.NewString()
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
> I think that's acceptable. 

Our tests show that we had a regression after upgrading the hub.docker.com. But OK, you know better.

> this is going to be less of a problem soon.

It will be less of a problem only then when there will be no schema1 on the servers.

> we need to fix right now is the tests

I know that now is broken: `TestImageStreamImport`, `TestImageStreamImportDockerHub`, but there could be more.